tiledb/sm/array_schema/attribute.cc

```diff
@@ -81,25 +81,11 @@ Attribute::Attribute(
     , name_(name)
     , type_(type)
     , order_(order) {
-  set_default_fill_value();
-
-  // If ordered, check the number of values of cells is supported.
   if (order_ != DataOrder::UNORDERED_DATA) {
     ensure_ordered_attribute_datatype_is_valid(type_);
-    if (type == Datatype::STRING_ASCII) {
-      if (cell_val_num_ != constants::var_num) {
-        throw std::invalid_argument(
-            "Ordered attributes with datatype '" + datatype_str(type_) +
-            "' must have cell_val_num=1.");
-      }
-    } else {
-      if (cell_val_num_ != 1) {
-        throw std::invalid_argument(
-            "Ordered attributes with datatype '" + datatype_str(type_) +
-            "' must have cell_val_num=1.");
-      }
-    }
   }
+  check_cell_val_num(cell_val_num_);
```

Comment:
It looks as if the constructor this statement is called from is not called anywhere. At least that's what code analysis is telling me; I haven't tried removing it. See if the whole constructor can go.

Reply:
Removed it and had to wade through a bit of template soup to find the call site, but it does look like this constructor is used in at least one place - `dimension_label.cc`